service/controller/resource/spark/create.go

```diff
@@ -296,13 +342,310 @@ func secretName(base string) string {
 	return base + "-machine-pool-ignition"
 }
 
-// toAzureConfig performs mapping from CAPI & CAPZ types to AzureConfig which
-// fulfills k8scloudconfig requirements for rendering node pool worker node
-// ignition templates.
-func toAzureConfig(cluster *capiv1alpha3.Cluster, azureCluster *capzv1alpha3.AzureCluster, machinePool *expcapiv1alpha3.MachinePool, azureMachinePool *expcapzv1alpha3.AzureMachinePool) providerv1alpha1.AzureConfig {
-	ac := providerv1alpha1.AzureConfig{}
+func getAvailabilityZones(machinePool expcapiv1alpha3.MachinePool) ([]int, error) {
+	var availabilityZones []int
+	for _, fd := range machinePool.Spec.FailureDomains {
+		intFd, err := strconv.Atoi(fd)
+		if err != nil {
+			return availabilityZones, microerror.Mask(err)
+		}
+		availabilityZones = append(availabilityZones, intFd)
+	}
+
+	return availabilityZones, nil
+}
+
+func (r *Resource) buildAzureConfig(ctx context.Context, cluster *capiv1alpha3.Cluster, azureCluster *capzv1alpha3.AzureCluster, machinePool *expcapiv1alpha3.MachinePool, azureMachinePool *expcapzv1alpha3.AzureMachinePool) (providerv1alpha1.AzureConfig, error) {
```

Comment:
I copied the code from the `azureconfig` handler that @tuommaki added to create `AzureConfig` crs from CAPI crs.